authentik/providers/saml/tasks.py

```diff
@@ -0,0 +1,116 @@
+"""SAML Provider tasks"""
+
+import requests
+from django.contrib.auth import get_user_model
+from django.http import HttpResponseBadRequest
+from django.utils.timezone import now
+from dramatiq.actor import actor
+from structlog.stdlib import get_logger
+
+from authentik.events.models import Event, EventAction
+from authentik.providers.saml.models import SAMLProvider, SAMLSession
+from authentik.providers.saml.processors.logout_request import LogoutRequestProcessor
+
+LOGGER = get_logger()
+User = get_user_model()
+
+
+@actor(description="Send SAML LogoutRequest to a Service Provider")
+def send_saml_logout_request(
+    provider_pk: int,
+    sls_url: str,
+    name_id: str,
+    name_id_format: str,
+    session_index: str,
+) -> bool:
+    """Send SAML LogoutRequest to a Service Provider using session data"""
+    try:
+        provider = SAMLProvider.objects.get(pk=provider_pk)
+
+        LOGGER.debug(
+            "Sending SAML logout request",
+            provider=provider.name,
+            session_index=session_index,
+        )
+
+        # Create the logout request processor with the provided data
+        # Note: We don't need the user object for the logout request itself
+        processor = LogoutRequestProcessor(
+            provider=provider,
+            user=None,  # User might be deleted already
+            destination=sls_url,
+            name_id=name_id,
+            name_id_format=name_id_format,
+            session_index=session_index,
+        )
+
+        success = send_post_logout_request(provider, processor)
+
+        # Log the event (without user reference since they might be deleted)
+        Event.new(
```

Comment:
Maybe for now let's not do that until we can figure out a way to get this data into the users' actual logout event